runtimes/eoapi/stac/eoapi/stac/client.py

```diff
@@ -413,39 +555,161 @@ async def get_item(
                 request.headers.get("accept", ""), accepted_media
             )
 
+        # Check if collection exist
+        await self.get_collection(collection_id, request=request)
+
+        search_request = self.pgstac_search_model(
+            ids=[item_id], collections=[collection_id], limit=1
+        )
+        item_collection = await self._search_base(search_request, request=request)
+        if not item_collection["features"]:
+            raise NotFoundError(
+                f"Item {item_id} in Collection {collection_id} does not exist."
+            )
+
         if output_type == MimeTypes.html:
             return create_html_response(
                 request,
-                item,
+                item_collection["features"][0],
                 template_name="item",
                 title=f"{collection_id}/{item_id} item",
             )
 
-        return item
+        return Item(**item_collection["features"][0])
 
     async def get_search(
         self,
         request: Request,
-        *args,
+        collections: Optional[List[str]] = None,
+        ids: Optional[List[str]] = None,
+        bbox: Optional[BBox] = None,
+        intersects: Optional[str] = None,
+        datetime: Optional[str] = None,
+        limit: Optional[int] = None,
+        # Extensions
+        query: Optional[str] = None,
+        fields: Optional[List[str]] = None,
+        sortby: Optional[str] = None,
+        filter_expr: Optional[str] = None,
+        filter_lang: Optional[str] = None,
+        token: Optional[str] = None,
         f: Optional[str] = None,
         **kwargs,
     ) -> ItemCollection:
-        items = await super().get_search(request, *args, **kwargs)
-
         output_type: Optional[MimeTypes]
         if f:
             output_type = MimeTypes[f]
         else:
-            accepted_media = [MimeTypes[v] for v in get_args(GeoResponseType)]
+            accepted_media = [MimeTypes[v] for v in get_args(GeoMultiResponseType)]
             output_type = accept_media_type(
                 request.headers.get("accept", ""), accepted_media
             )
 
+        # Parse request parameters
+        base_args = {
+            "collections": collections,
+            "ids": ids,
+            "bbox": bbox,
+            "limit": limit,
+            "token": token,
+            "query": orjson.loads(unquote_plus(query)) if query else query,
+        }
+
+        clean = self._clean_search_args(
+            base_args=base_args,
+            intersects=intersects,
+            datetime=datetime,
+            fields=fields,
+            sortby=sortby,
+            filter_query=filter_expr,
+            filter_lang=filter_lang,
+        )
+
+        search_request = self.pgstac_search_model(**clean)
+        item_collection = await self._search_base(search_request, request=request)
```

Comment:
the original `get_search` method calls `post_search` in https://github.com/stac-utils/stac-fastapi-pgstac/blob/3f1ffa37443a1a1e3b271e05f614fbc6766ef4e1/stac_fastapi/pgstac/core.py#L511

Because of the possible `JSONResponse` (fields), we need to re-write the whole method ourself